src/NuGet.Core/NuGet.Packaging/Signing/TrustStore/X509TrustStore.cs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using NuGet.Common;
+
+namespace NuGet.Packaging.Signing
+{
+    /// <summary>
+    /// Represents the X.509 trust store that package signing and signed package verification will use.
+    /// </summary>
+    public static class X509TrustStore
+    {
+        private static IX509ChainFactory Instance;
+        private static readonly object LockObject = new();
+
+        internal static bool IsEnabled { get; }
+
+        static X509TrustStore()
+        {
+            Version DotNet7 = new(major: 7, minor: 0);
+
+            IsEnabled = Environment.Version >= DotNet7;
+        }
+
+        /// <summary>
+        /// Attempts to discover and load an X.509 trust store and log details about the attempt.
+        /// If initialization has already happened, a call to this method will have no effect.
+        /// </summary>
+        /// <remarks>Explicit initialization is unnecessary for X.509 certificate chain building;
+        /// initialization will happen automatically on demand.  This method provides an opportunity
+        /// to log diagnostic information about trust store discovery.</remarks>
+        /// <param name="logger">A logger.</param>
+        /// <exception cref="ArgumentNullException">Thrown if <paramref name="logger" /> is <c>null</c>.</exception>
+        public static void Initialize(ILogger logger)
+        {
+            if (logger is null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            _ = GetX509ChainFactory(logger);
+        }
+
+        internal static IX509ChainFactory GetX509ChainFactory(ILogger logger)
+        {
+            if (logger is null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            if (Instance is not null)
```

Comment:
What do you think about using `Lazy<T>` here instead for a singleton?

Reply:
Initialization is dependent on method argument `logger`.  If you try using `Lazy<T>`, I think you'll agree that the solution is more complicated than what is already here.

We want the semantics similar to **`LazyThreadSafetyMode.ExecutionAndPublication` but without having to also deal with lazy initialization of the `Lazy<T>` instance itself.

If you feel you have a better solution, contact me offline to discuss.